File: workplace-fe/src/app/pages/profile/builder/my-decks/profile-deck.ts
export class ProfileDeck {
  id: string | undefined;
  profileId: string | undefined;
  title: string | undefined;
  info: string | undefined;
  decks: ProfileDeckItem[] = [];
}

export class ProfileDeckItem {
  id: string | undefined;
  image: string | undefined;
  title: string | undefined;
  info: string | undefined;
}

File: workplace-be/src/main/java/com/emetaplus/workplace/builder/decks/model/ProfileDeck.java
package com.emetaplus.workplace.builder.decks.model;

import com.emetaplus.workplace.profile.model.Profile;
import lombok.Data;

import javax.persistence.*;
import java.util.List;
import java.util.UUID;

@Data
@Entity
@Table(name = "profile_decks")
public class ProfileDeck {

    @Id
    @GeneratedValue
    @Column(name = "profile_deck_id")
    private UUID id;

    @Column
    private String title;

    @Column
    private String info;

    @OneToOne
    @JoinColumn(name = "profile_id", unique = true, nullable = false)
    private Profile profile;

    @OneToMany(cascade = CascadeType.ALL, orphanRemoval=true, fetch = FetchType.EAGER)
    @JoinColumn(name = "profile_deck_id", nullable = false, insertable = false, updatable = false)
    private List<ProfileDeckItem> decks;

}
